dvc/remote/base.py

```diff
@@ -218,9 +217,10 @@ def make_names(hash_, names):
 
         return dir_status, file_status, dir_contents
 
-    def _indexed_dir_hashes(self, cache, named_cache, dir_md5s):
+    def _indexed_dir_hashes(self, dir_objs):
         # Validate our index by verifying all indexed .dir hashes
         # still exist on the remote
+        dir_md5s = set(dir_objs.keys())
```

Comment:
all of these `dir_md5`/`file_md5` sets/dicts should be dropped in favor of just using the tree objects, but I don't want to touch this functionality yet in this PR